.github/workflows/cd_release.yml

```diff
@@ -288,15 +290,21 @@ jobs:
     - name: Get tagged versions
       run: |
         if [ -n "${{ inputs.changelog_exclude_tags_regex }}" ]; then
-          echo "PREVIOUS_VERSION=$(git tag -l --sort -version:refname | grep -v -E "${{ inputs.changelog_exclude_tags_regex }}" | sed -n 2p)" >> $GITHUB_ENV
+          PREVIOUS_VERSION="$(git tag -l --sort -version:refname | grep -v -E "${{ inputs.changelog_exclude_tags_regex }}" | sed -n 2p)"
+        else
+          PREVIOUS_VERSION="$(git tag -l --sort -version:refname | sed -n 2p)"
+        fi
+
+        if [ -n "${PREVIOUS_VERSION}" ]; then
```

Comment:
I would suggest that in this if statement you define SINCE_PREVIOUS_VERSION and outside the if statement you append it to the GITHUB_ENV, you could even define SINCE_PERVIOUS_VERSION='' to be the default and then append --since-tag "${PREVIOUS_VERSION} --> this would eliminate the need of the else statment, then at the end you run echo 'SINCE_PREVIOUS_VERSION=${SINCE_PREVIOUS_VERSION}' >> $GITHUB_ENV.

Reply:
Sure. It's all the same thing in the end.

The old "to else or not to else" is an ever recurring question and consideration, and I'm constantly going back and forth between the two options. In my opinion, with the 'else', it becomes slightly more understandable, but a bit bulkier. Without the 'else' it feels "safer" due to the variable already being set (something that is without consequence in bash, though) and a bit "nicer", since only a single indentation block is necessary.

I think I'll look at what I've done in the rest of the file in these cases and do similarly here for consistency.

Adding the variable to the `$GITHUB_ENV` variable after the if-logic might be nicer, only if to make the whole thing more readable. Although, it will require an extra line of code as well as an extra use of a variable - something that is utterly without consequence in this case, and readability (or rather, understandability) should definitely be prioritized for all the workflow files.

I think I will do it here and make an issue to follow up on this in the whole code base - an issue that should be able to be picked up by anyone :)